### PR TITLE
many: rename registries to confdbs

### PR DESCRIPTION
**Pull request by miguelpires**
Commit [1136f37](https://github.com/canonical/snapd/pull/14768/commits/1136f37e12128a3819b27c4d0e9d4932812f35c0) renames the registries feature to confdbs. This includes the assertions and feature flags. So anyone who has the experimental feature enabled should unset it, refresh and then it enable it back (otherwise, change the state and exported file directly if you can do that safely). Assertions will need to be regenerated with the new assertion type.
Commit [11f18d9](https://github.com/canonical/snapd/pull/14768/commits/11f18d9e1fb150029207f7c155f545499b882712) pluralizes the confdb API endpoint, renaming it form /v2/confdb/... to /v2/confdbs/... which is unrelated to the previous commit but something I've been meaning to do anyway so might as well change it once

### CODE REVIEW

**Review by andrewphelpsj — APPROVED** [review]
Thank you!

File: interfaces/builtin/confdb.go
Comment (ZeyadYasser) [inline interfaces/builtin/confdb.go:1]: I think this warrants a change to the exclusion map in `tests/main/document-interfaces-url/task.yaml`? I am confused by why it didn't fail in CI.

File: overlord/assertstate/assertstate.go [hunk 1]
```diff
@@ -390,34 +390,34 @@ func AutoRefreshAssertions(s *state.State, userID int) error {
...
 
-// autoRefreshRegistryAssertions fetches the newest revision of all stored
-// registry assertions.
-func autoRefreshRegistryAssertions(st *state.State, userID int, opts *RefreshAssertionsOptions) error {
+// autoRefreshConfdbAssertions fetches the newest revision of all stored
+// confdb assertions.
+func autoRefreshConfdbAssertions(st *state.State, userID int, opts *RefreshAssertionsOptions) error {
 	db := cachedDB(st)
-	regAsserts, err := db.FindMany(asserts.RegistryType, nil)
+	regAsserts, err := db.FindMany(asserts.ConfdbType, nil)
```
Comment (ZeyadYasser) [inline overlord/assertstate/assertstate.go:400]:
```suggestion
	confdbAsserts, err := db.FindMany(asserts.ConfdbType, nil)
```

File: overlord/assertstate/assertstate.go [hunk 2]
```diff
@@ -390,34 +390,34 @@ func AutoRefreshAssertions(s *state.State, userID int) error {
...
 	if err != nil {
 		if errors.Is(err, &asserts.NotFoundError{}) {
 			return nil
 		}
 		return err
 	}
 
-	var registries []*registry.Registry
+	var confdbs []*confdb.Confdb
 	for _, regAs := range regAsserts {
```
Comment (ZeyadYasser) [inline overlord/assertstate/assertstate.go:409]:
```suggestion
	for _, confdbAs := range confdbAsserts {
```

File: overlord/assertstate/assertstate.go [hunk 3]
```diff
@@ -390,34 +390,34 @@ func AutoRefreshAssertions(s *state.State, userID int) error {
...
-		reg := regAs.(*asserts.Registry).Registry()
-		registries = append(registries, reg)
+		reg := regAs.(*asserts.Confdb).Confdb()
```
Comment (ZeyadYasser) [inline overlord/assertstate/assertstate.go:410]:
```suggestion
		confdb := confdbAs.(*asserts.Confdb).Confdb()
```

File: overlord/assertstate/assertstate_test.go [hunk 4]
```diff
@@ -5492,7 +5492,7 @@ func (s *assertMgrSuite) setupRegistry(c *C) *snap.SideInfo {
     }
   }
 }`
-	regAs := s.registry(c, "my-registry", extraHeaders, schema)
+	regAs := s.confdb(c, "my-confdb", extraHeaders, schema)
```
Comment (ZeyadYasser) [inline overlord/assertstate/assertstate_test.go:5495]:
```suggestion
	confdbAs := s.confdb(c, "my-confdb", extraHeaders, schema)
```

File: overlord/assertstate/assertstate_test.go [hunk 5]
```diff
@@ -5544,9 +5544,9 @@ func (s *assertMgrSuite) TestFetchRegistryAssertion(c *C) {
 
 	c.Assert(chg.Err(), IsNil)
 
-	reg, err := assertstate.DB(s.state).Find(asserts.RegistryType, map[string]string{
+	reg, err := assertstate.DB(s.state).Find(asserts.ConfdbType, map[string]string{
```
Comment (ZeyadYasser) [inline overlord/assertstate/assertstate_test.go:5547]:
```suggestion
	confdb, err := assertstate.DB(s.state).Find(asserts.ConfdbType, map[string]string{
```

File: overlord/assertstate/assertstate_test.go [hunk 6]
```diff
@@ -5603,11 +5603,11 @@ func (s *assertMgrSuite) testRegistryAssertionsAutoRefresh(c *C) {
     }
   }
 }`
-	regAs := s.registry(c, "my-registry", extraHeaders, schema)
+	regAs := s.confdb(c, "my-confdb", extraHeaders, schema)
```
Comment (ZeyadYasser) [inline overlord/assertstate/assertstate_test.go:5606]:
```suggestion
	confdbAs := s.confdb(c, "my-confdb", extraHeaders, schema)
```

File: overlord/assertstate/assertstate_test.go [hunk 7]
```diff
@@ -5616,24 +5616,24 @@ func (s *assertMgrSuite) testRegistryAssertionsAutoRefresh(c *C) {
 	// precondition check
 	c.Assert(assertstate.AutoRefreshAssertions(s.state, 0), IsNil)
 	db := assertstate.DB(s.state)
-	reg, err := db.Find(asserts.RegistryType, map[string]string{
+	reg, err := db.Find(asserts.ConfdbType, map[string]string{
```
Comment (ZeyadYasser) [inline overlord/assertstate/assertstate_test.go:5619]:
```suggestion
	confdb, err := db.Find(asserts.ConfdbType, map[string]string{
```

File: overlord/assertstate/assertstate_test.go [hunk 8]
```diff
@@ -5616,24 +5616,24 @@ func (s *assertMgrSuite) testRegistryAssertionsAutoRefresh(c *C) {
...
 		"account-id": s.dev1Acct.AccountID(),
-		"name":       "my-registry",
+		"name":       "my-confdb",
 	})
 	c.Assert(err, IsNil)
 	c.Check(reg.Revision(), Equals, 1)
 
 	extraHeaders["revision"] = "2"
-	regAs = s.registry(c, "my-registry", extraHeaders, schema)
+	regAs = s.confdb(c, "my-confdb", extraHeaders, schema)
```
Comment (ZeyadYasser) [inline overlord/assertstate/assertstate_test.go:5627]:
```suggestion
	confdbAs = s.confdb(c, "my-confdb", extraHeaders, schema)
```

File: overlord/confdbstate/confdbmgr_test.go [hunk 9]
```diff
@@ -54,16 +54,16 @@ func (s *hookHandlerSuite) SetUpTest(c *C) {
 	s.repo = interfaces.NewRepository()
 	ifacerepo.Replace(s.state, s.repo)
 
-	regIface := &ifacetest.TestInterface{InterfaceName: "registry"}
+	regIface := &ifacetest.TestInterface{InterfaceName: "confdb"}
```
Comment (ZeyadYasser) [inline overlord/confdbstate/confdbmgr_test.go:57]:
```suggestion
	confdbIface := &ifacetest.TestInterface{InterfaceName: "confdb"}
```

File: overlord/confdbstate/confdbstate_test.go [hunk 10]
```diff
@@ -389,15 +389,15 @@ func (s *registryTestSuite) TestPlugsAffectedByPaths(c *C) {
...
+	}, confdb.NewJSONSchema())
 	c.Assert(err, IsNil)
 
 	repo := interfaces.NewRepository()
 	s.state.Lock()
 	defer s.state.Unlock()
 	ifacerepo.Replace(s.state, repo)
 
-	regIface := &ifacetest.TestInterface{InterfaceName: "registry"}
+	regIface := &ifacetest.TestInterface{InterfaceName: "confdb"}
```
Comment (ZeyadYasser) [inline overlord/confdbstate/confdbstate_test.go:400]:
```suggestion
	confdbIface := &ifacetest.TestInterface{InterfaceName: "confdb"}
```

File: overlord/confdbstate/confdbstate_test.go [hunk 11]
```diff
@@ -612,57 +612,57 @@ func (s *registryTestSuite) TestRegistryTasksObserverSnapSetWithCustodianInstall
...
+	c.Assert(err, ErrorMatches, fmt.Sprintf("cannot commit changes to confdb %s/network: no custodian snap installed", s.devAccID))
 }
 
-func (s *registryTestSuite) setupRegistryModificationScenario(c *C, custodians, nonCustodians []string) {
+func (s *confdbTestSuite) setupConfdbModificationScenario(c *C, custodians, nonCustodians []string) {
 	s.repo = interfaces.NewRepository()
 	ifacerepo.Replace(s.state, s.repo)
 
-	regIface := &ifacetest.TestInterface{InterfaceName: "registry"}
+	regIface := &ifacetest.TestInterface{InterfaceName: "confdb"}
```
Comment (ZeyadYasser) [inline overlord/confdbstate/confdbstate_test.go:655]:
```suggestion
	confdbIface := &ifacetest.TestInterface{InterfaceName: "confdb"}
```

File: overlord/confdbstate/confdbstate_test.go [hunk 12]
```diff
@@ -1109,27 +1109,27 @@ func (s *registryTestSuite) testGetReadableOngoingTransaction(c *C, hook string)
...
+	commitTask.Set("confdb-transaction", tx)
 
 	refTask := s.state.NewTask("change-view-setup", "")
 	chg.AddTask(refTask)
 	refTask.Set("commit-task", commitTask.ID())
 
-	// make some other registry to access concurrently
-	reg, err := registry.New("foo", "bar", map[string]interface{}{
+	// make some other confdb to access concurrently
+	reg, err := confdb.New("foo", "bar", map[string]interface{}{
```
Comment (ZeyadYasser) [inline overlord/confdbstate/confdbstate_test.go:1128]:
```suggestion
	db, err := confdb.New("foo", "bar", map[string]interface{}{
```

File: overlord/hookstate/ctlcmd/get_test.go [hunk 13]
```diff
@@ -872,40 +872,40 @@ func (s *registrySuite) TestRegistryGetDifferentViewThanOngoingTx(c *C) {
...
+	// set ongoing tx related to the network confdb
+	task.Set("confdb-transaction", tx)
 
 	s.state.Unlock()
 	defer s.state.Lock()
 
-	restore := ctlcmd.MockRegistrystateGetView(func(st *state.State, account, registryName, viewName string) (*registry.View, error) {
-		reg, err := registry.New(s.devAccID, "other", map[string]interface{}{
+	restore := ctlcmd.MockConfdbstateGetView(func(st *state.State, account, confdbName, viewName string) (*confdb.View, error) {
+		reg, err := confdb.New(s.devAccID, "other", map[string]interface{}{
```
Comment (ZeyadYasser) [inline overlord/hookstate/ctlcmd/get_test.go:882]:
```suggestion
		db, err := confdb.New(s.devAccID, "other", map[string]interface{}{
```

File: overlord/confdbstate/transaction_test.go [hunk 14]
```diff
@@ -168,22 +168,22 @@ func (s *transactionTestSuite) TestRollBackOnCommitError(c *C) {
 }
 
 func (s *transactionTestSuite) TestManyWrites(c *C) {
-	tx, err := registrystate.NewTransaction(s.state, "my-account", "my-reg")
+	tx, err := confdbstate.NewTransaction(s.state, "my-account", "my-reg")
```
Comment (ZeyadYasser) [inline overlord/confdbstate/transaction_test.go:171]: nitpick, there are multiple instances of "my-reg"

**Review by ZeyadYasser — APPROVED** [review]
LGTM! only small nitpicks

File: interfaces/builtin/confdb.go
Comment (miguelpires) [reply interfaces/builtin/confdb.go:1]: My bad, I did that in another PR that changed that test by mistake and didn't fix it here. It also didn't fail there strangely, I'm guessing that it got a 500 instead but haven't validated this yet https://github.com/canonical/snapd/pull/14777#discussion_r1862459776

**Comment by ernestl** [comment]
Failures:
- centos, fedora, opensuse | known issues
- [spread nested-ubuntu-24.04](https://github.com/canonical/snapd/actions/runs/12157220103/job/33911675086?pr=14768#logs) | known issue

**Pull request merged**